stdlib/public/runtime/ImageInspectionELF.cpp

```diff
@@ -20,32 +20,37 @@
 
 #if defined(__ELF__)
 
+#include "../SwiftShims/MetadataSections.h"
 #include "ImageInspection.h"
 #include "ImageInspectionELF.h"
 #include <dlfcn.h>
 
 using namespace swift;
 
 namespace {
-static const swift::MetadataSections *registered = nullptr;
+static swift::MetadataSections *registered = nullptr;
 
 void record(const swift::MetadataSections *sections) {
+  // We cast off the const in order to update the linked list
+  // data structure. This is safe to do since we don't touch 
+  // any other fields.
+  auto casted_sections = const_cast<MetadataSections *>(sections);
```

Comment:
What about making the parameter non-const, and cast away the `const` prior to invoking the `record` method?

Reply:
@compnerd Updated!